File: features/advisor/src/main/java/com/nivasafinance/features/advisorlead/exception/AdvisorLeadMappingExceptionFactory.java
package com.nivasafinance.features.advisorlead.exception;

import org.springframework.context.MessageSource;

import java.util.UUID;

public class AdvisorLeadMappingExceptionFactory {

    public static AdvisorLeadMappingNotFoundException notFound(UUID id, MessageSource messageSource) {
        return new AdvisorLeadMappingNotFoundException(id, messageSource);
    }

    public static AdvisorLeadMappingOperationException createFailed(MessageSource messageSource) {
        return new AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.create", messageSource);
    }

    public static AdvisorLeadMappingOperationException updateFailed(MessageSource messageSource) {
        return new AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.update", messageSource);
    }

    public static AdvisorLeadMappingOperationException deleteFailed(MessageSource messageSource) {
        return new AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.delete", messageSource);
    }

    public static AdvisorLeadMappingOperationException retrieveEntityFailed(MessageSource messageSource) {
        return new AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.retrieve", messageSource);
    }
}

File: features/advisorleadmapping/src/main/java/com/nivasafinance/features/advisorleadmapping/exception/AdvisorLeadMappingExceptionFactory.kt
package com.nivasafinance.features.advisorleadmapping.exception

import org.springframework.context.MessageSource
import java.util.UUID

object AdvisorLeadMappingExceptionFactory {

    fun notFound(id: UUID, messageSource: MessageSource): AdvisorLeadMappingNotFoundException {
        return AdvisorLeadMappingNotFoundException(id, messageSource)
    }

    fun createFailed(messageSource: MessageSource): AdvisorLeadMappingOperationException {
        return AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.create", messageSource)
    }

    fun updateFailed(messageSource: MessageSource): AdvisorLeadMappingOperationException {
        return AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.update", messageSource)
    }

    fun deleteFailed(messageSource: MessageSource): AdvisorLeadMappingOperationException {
        return AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.delete", messageSource)
    }

    fun retrieveEntityFailed(messageSource: MessageSource): AdvisorLeadMappingOperationException {
        return AdvisorLeadMappingOperationException("error.advisor.lead.mapping.operation.retrieve", messageSource)
    }
}
